src/rendering/SceneManager.cc

```diff
@@ -1585,6 +1585,9 @@ void SceneManager::RemoveEntity(Entity _id)
     auto it = this->dataPtr->visuals.find(_id);
     if (it != this->dataPtr->visuals.end())
     {
+      ignition::common::MeshManager *meshManager =
+          ignition::common::MeshManager::Instance();
+      meshManager->RemoveMesh(it->second->Name());
```

Comment:
The same mesh can be used by multiple entities. So it can be dangerous to remove the mesh when removing one of these entities. Have you tested this case?